### PR TITLE
[DPE-1697] Initial test app

### PR DESCRIPTION
**Pull request by dragomirp**
Initial version of the merged test charm. Should be usable by all PG and PGB charms.

Test PRs with the same test charm:
https://github.com/canonical/pgbouncer-operator/pull/113
https://github.com/canonical/pgbouncer-k8s-operator/pull/137
https://github.com/canonical/postgresql-operator/pull/215
https://github.com/canonical/postgresql-k8s-operator/pull/230

### CODE REVIEW

**Review by taurus-forever — APPROVED** [review]
Nice! BTW, we need to enable renovate here too :-D

**Comment by dragomirp** [comment]
Postgresql VM tests are still to pass, but I think that is due to refactoring within the charm and shouldn't require changes to the test app.

File: src/charm.py [hunk 1]
```diff
@@ -0,0 +1,502 @@
...
+            fd.write(self._connection_string)
+            os.fsync(fd)
+
+        try:
+            os.kill(int(self.app_peer_data[PROC_PID_KEY]), signal.SIGKILL)
+        except ProcessLookupError:
+            del self.app_peer_data[PROC_PID_KEY]
+            return
+        count = self._count_writes()
+        self._start_continuous_writes(count + 1)
```
Comment (dragomirp) [inline src/charm.py:185]: We need to take care of endpoint switching for HA in Postgresql VM.

**Review by marceloneppel — APPROVED** [review]
Great work, Dragomir! Thanks!

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,74 @@
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+[tool.poetry]
+name = "postgresql-k8s"
```
[the remark this answers is not in the document]
Comment (dragomirp) [reply pyproject.toml:5]: Yes, nice catch.

**Review by marceloneppel — APPROVED** [review]

**Comment by taurus-forever** [comment]
Feel free to merge and publish revsion 1, the charmhub part is ready: https://discourse.charmhub.io/t/request-change-ownership-for-postgresql-test-app/11690/2
Great job done!

**Pull request merged**